test/clj/game/cards/agendas_test.clj

```diff
@@ -448,6 +448,32 @@
    (is (= 0 (count (:discard (get-corp)))) "No card trashed")
    (is (= 0 (count (:scored (get-corp)))) "Azef Protocol not scored")))
 
+(deftest bacterial-programming-stolen-from-archives-should-not-give-duplicate-accesses
+  ;; there used to be an issue where you could randomly access the same cards indefinitely
+  (do-game
+    (new-game {:corp {:discard ["Hedge Fund" "Bacterial Programming"]
+                      :hand ["Bellona"]
+                      :deck ["IPO" "NGO Front" "Rashida Jaheem" "Tithe"
+                             "Ice Wall" "Fire Wall" "Enigma"]}})
+    (take-credits state :corp)
+    (run-empty-server state :archives)
+    (click-prompt state :runner "Bacterial Programming")
+    (click-prompt state :runner "Steal")
+    (click-prompt state :corp "Yes")
+    (doseq [_ (range 7)
```

Comment:
this is `dotimes [_ 7]`

Reply:
ah, true. 
I just mindlessly copied the other test :joy: 

Updated it on both, keeping the `TODO` in because it's something that should (eventually) be done if it hasn't already.